### PR TITLE
Remove URL credentials from incoming REPO_URL

### PR DESCRIPTION
**Pull request by bduffany**
---

**Version bump**: Patch

### CODE REVIEW

File: server/util/url/url.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package url
+
+import (
+	"regexp"
+	"strings"
+)
+
+var (
+	userPasswordRegex = regexp.MustCompile("^(https?://)(.*?:.*?@)")
```
Comment (siggisim) [inline server/util/url/url.go:9]: i think we should probably support more formats, like git://, ssh://, and one without a prefix:
https://stackoverflow.com/questions/31801271/what-are-the-supported-git-url-formats

Comment (siggisim) [reply server/util/url/url.go:9]: or you can go straight to the source: https://github.com/git/git/blob/77bd3ea9f54f1584147b594abc04c26ca516d987/t/t0110-urlmatch-normalization.sh

Comment (bduffany) [reply server/util/url/url.go]: Now using the `git-urls` library for parsing, with some additional logic layered on top to handle protocol-less URLs (e.g. `github.com/foo/bar`) and `localhost`

File: server/build_event_protocol/accumulator/accumulator.go [hunk 2]
```diff
@@ -6,6 +6,8 @@ import (
 
 	"github.com/buildbuddy-io/buildbuddy/proto/build_event_stream"
 	"github.com/buildbuddy-io/buildbuddy/proto/command_line"
+
+	gitutil "github.com/buildbuddy-io/buildbuddy/server/util/git"
```
Comment (tylerwilliams) [inline server/build_event_protocol/accumulator/accumulator.go:10]: we should name this file + package gitutil too

**Review by tylerwilliams — APPROVED** [review]
As long as this doesn't change what we accept in workflows, LGTM :)

(one nit)

File: server/build_event_protocol/accumulator/accumulator.go (hunk 2 above)
Comment (bduffany) [reply server/build_event_protocol/accumulator/accumulator.go:10]: OK, let me send that as a follow-up PR so it's an easy-LGTM

**Pull request merged**